shell/platform/darwin/macos/framework/Source/FlutterMetalCompositor.mm

```diff
@@ -94,13 +97,32 @@
         break;
       }
       case kFlutterLayerContentTypePlatformView:
-        // Add functionality in follow up PR.
-        FML_LOG(WARNING) << "Presenting PlatformViews not yet supported";
+        PresentPlatformView(layer, i);
         break;
     };
   }
 
   return EndFrame(has_flutter_content);
 }
 
+void FlutterMetalCompositor::PresentPlatformView(const FlutterLayer* layer, size_t layer_position) {
+  // TODO (https://github.com/flutter/flutter/issues/96668)
+  FML_DCHECK([[NSThread currentThread] isMainThread])
+      << "Must be on the main thread to handle presenting platform views";
+
+  int64_t platform_view_id = layer->platform_view->identifier;
+  NSView* platform_view = [platform_views_controller_ platformViewWithID:platform_view_id];
+
+  FML_DCHECK(platform_view) << "Platform view not found for id: " << platform_view_id;
+
+  CGFloat scale = [[NSScreen mainScreen] backingScaleFactor];
+  platform_view.frame = CGRectMake(layer->offset.x / scale, layer->offset.y / scale,
+                                   layer->size.width / scale, layer->size.height / scale);
+  if (platform_view.superview == nil) {
+    [view_controller_.flutterView addSubview:platform_view];
+  } else {
+    platform_view.layer.zPosition = layer_position;
```

Comment:
Maybe we can set the zPosition even for newly added platform_view just to ensure zPositions to be correct for each frame:
```objc
  if (platform_view.superview == nil) {
    [view_controller_.flutterView addSubview:platform_view];
  }
  platform_view.layer.zPosition = layer_position;
```

Reply:
Done!